primefaces/src/main/java/org/primefaces/validate/FileValidator.java

```diff
@@ -0,0 +1,323 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.primefaces.validate;
+
+import org.apache.poi.util.StringUtil;
+import org.primefaces.component.fileupload.FileUpload;
+import org.primefaces.context.PrimeApplicationContext;
+import org.primefaces.model.file.UploadedFile;
+import org.primefaces.model.file.UploadedFiles;
+import org.primefaces.util.FileUploadUtils;
+import org.primefaces.util.LangUtils;
+import org.primefaces.util.MessageFactory;
+import org.primefaces.virusscan.VirusException;
+
+import javax.faces.application.FacesMessage;
+import javax.faces.component.PartialStateHolder;
+import javax.faces.component.UIComponent;
+import javax.faces.context.FacesContext;
+import javax.faces.validator.Validator;
+import javax.faces.validator.ValidatorException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class FileValidator implements Validator, PartialStateHolder, ClientValidator {
+
+    private static final String VALIDATOR_ID = "primefaces.File";
+    private static final String FILE_LIMIT_MESSAGE_ID = "primefaces.FileValidator.FILE_LIMIT";
+    private static final String ALLOW_TYPES_MESSAGE_ID = "primefaces.FileValidator.ALLOW_TYPES";
+    private static final String SIZE_LIMIT_MESSAGE_ID = "primefaces.FileValidator.SIZE_LIMIT";
+
+    private Integer fileLimit;
+    private String fileLimitMessage;
+    private Long sizeLimit;
+    private String sizeLimitMessage;
+    private String allowTypes;
+    private String allowTypesMessage;
+
+    private Boolean virusScan;
+
+    private boolean isTransient = false;
+    private boolean initialStateMarked = false;
+
+    @Override
+    public void validate(FacesContext context, UIComponent component, Object value) throws ValidatorException {
+
+        if (component instanceof FileUpload) {
+            FileUpload fileUpload = (FileUpload) component;
+
+            if (value instanceof UploadedFile) {
+                UploadedFile uploadedFile = (UploadedFile) value;
+
+                validateUploadedFile(context, fileUpload, uploadedFile);
+            }
+            else if (value instanceof UploadedFiles) {
+                UploadedFiles uploadedFiles = (UploadedFiles) value;
+
+                validateUploadedFiles(context, fileUpload, uploadedFiles);
+            }
+        }
```

Comment:
throw exception if not FileUpload?

Reply:
i will try to implement h:inputFile support, otherwise will do it !